app/src/main/java/com/example/drawingactivity/DrawActivity.kt

```diff
@@ -42,17 +45,23 @@ class DrawActivity : AppCompatActivity() {
         colorSelector.onChangeColor = { viewModel.cycleCurrentColor() }
         val coordinator = findViewById<CoordinatorLayout>(R.id.coordinator)
         val bottomSheet = coordinator.findViewById<LinearLayout>(R.id.toolLayout)
-        behavior = BottomSheetBehavior.from(bottomSheet)
-        listOf<View>(
+        val viewList = listOf<View>(
             bottomSheet.findViewById<CurrentColorView>(R.id.pickerButton),
             bottomSheet.findViewById<ImageView>(R.id.brushButton),
             bottomSheet.findViewById<ImageView>(R.id.moveButton),
             bottomSheet.findViewById<ImageView>(R.id.resizeButton),
             bottomSheet.findViewById<ImageView>(R.id.filterButton),
-            bottomSheet.findViewById<ImageView>(R.id.layerButton),
-        ).forEach {
-            it.setOnClickListener { behavior.state = BottomSheetBehavior.STATE_EXPANDED }
-        }
+            bottomSheet.findViewById<ImageView>(R.id.layerButton)
+        )
+        val layoutList = listOf<View>(
```

Comment:
I bet this listOf was telling you it didn't need the generic any more cause it recognizes all the elements are Views 🤔

Reply:
it actually doesn't give mem a tip or anything for that, but yeah I knew I needed a reference somehow, I'll try the data class approach